megamek/src/megamek/common/AmmoType.java

```diff
@@ -15897,17 +15897,16 @@ public static boolean canSwitchToAmmo(Mounted weapon, AmmoType otherAmmo) {
         
         AmmoType currentAmmoType = (AmmoType) weapon.getLinked().getType();
         
-        boolean ammoOfSameType = currentAmmoType.equals(otherAmmo);
+        boolean ammoOfSameType = currentAmmoType.equalsAmmoTypeOnly(otherAmmo)
+                && (currentAmmoType.getRackSize() == otherAmmo.getRackSize());
         
         // MMLs can swap between different specific ammo types, so we have a special case check here
-        boolean mmlAmmoMatch = currentAmmoType.getAmmoType() == AmmoType.T_MML &&
-                otherAmmo.getAmmoType() == AmmoType.T_MML &&
-                currentAmmoType.getRackSize() == otherAmmo.getRackSize();
+        boolean mmlAmmoMatch = (currentAmmoType.getAmmoType() == AmmoType.T_MML)
+                && (otherAmmo.getAmmoType() == AmmoType.T_MML);
         
         // LBXs can swap between cluster and slug ammo types
-        boolean lbxAmmoMatch = currentAmmoType.getAmmoType() == AmmoType.T_AC_LBX &&
-                otherAmmo.getAmmoType() == AmmoType.T_AC_LBX &&
-                currentAmmoType.getRackSize() == otherAmmo.getRackSize();
+        boolean lbxAmmoMatch = (currentAmmoType.getAmmoType() == AmmoType.T_AC_LBX)
+                && (otherAmmo.getAmmoType() == AmmoType.T_AC_LBX);
```

Comment:
Can't look at it right now, what's the final logic block look like at the bottom that's returned? The one that's like "(mmlAmmoMatch || lbxAmmoMatch || ammoOfSameType) && !mismatch

Reply:
You can't even view it on GH its too large...oy.
```
    /**
     * Whether the given weapon can switch to the given ammo type
     * @param weapon The weapon being considered
     * @param otherAmmo The other ammo type being considered
     * @return true/false - null arguments or linked ammo bin for the weapon result in false
     */
    public static boolean canSwitchToAmmo(Mounted weapon, AmmoType otherAmmo) {
        if((weapon == null) ||
                (weapon.getLinked() == null) ||
                (otherAmmo == null)) {
            return false;
        }
        
        AmmoType currentAmmoType = (AmmoType) weapon.getLinked().getType();
        
        boolean ammoOfSameType = currentAmmoType.equalsAmmoTypeOnly(otherAmmo)
                && (currentAmmoType.getRackSize() == otherAmmo.getRackSize());
        
        // MMLs can swap between different specific ammo types, so we have a special case check here
        boolean mmlAmmoMatch = (currentAmmoType.getAmmoType() == AmmoType.T_MML)
                && (otherAmmo.getAmmoType() == AmmoType.T_MML);
        
        // LBXs can swap between cluster and slug ammo types
        boolean lbxAmmoMatch = (currentAmmoType.getAmmoType() == AmmoType.T_AC_LBX)
                && (otherAmmo.getAmmoType() == AmmoType.T_AC_LBX);
        
        boolean caselessLoaded = currentAmmoType.getMunitionType() == AmmoType.M_CASELESS;
        boolean otherBinCaseless = otherAmmo.getMunitionType() == AmmoType.M_CASELESS;
        boolean caselessMismatch = caselessLoaded != otherBinCaseless;
        
        boolean hasStaticFeed = weapon.hasQuirk(OptionsConstants.QUIRK_WEAP_NEG_STATIC_FEED);
        boolean staticFeedMismatch = hasStaticFeed && (currentAmmoType.getMunitionType() != otherAmmo.getMunitionType());
        
        return (ammoOfSameType || mmlAmmoMatch || lbxAmmoMatch) && !caselessMismatch && !staticFeedMismatch;
    }
```

Reply:
Hell I missed the rack size fix, let me add that back.